app/src/main/java/com/example/mytestproject/ui/weatherData/dailyWeather/DailyWeatherViewModel.kt

```diff
@@ -43,7 +45,15 @@ class DailyWeatherViewModel(
     }
 
     fun getWeatherData(days: Int) { // the method takes the number of days that come from the fragment
-        fetchDailyWeather(days)
+        numberOfWeatherForecastDays = days
+        fetchDailyWeather()
+    }
+
+    fun refreshWeatherDataList(swipeRefreshLayout: SwipeRefreshLayout) {
```

Comment:
We shouldn't pass any view to ViewModel. ViewMoldel and Fragment have different lifecycle. It can lead to memory leak.
This code should be done in Fragment. ViewModel only controls the value of swipeRefreshLayout.isRefreshing flag by data binding.

Reply:
Fixed